src/parser/mod.rs

```diff
@@ -49,6 +54,15 @@ pub struct MetaValue {
     pub opaque_token: Option<Vec<u8>>,
 }
 
+/// Memcached value generated by either an ASCII or meta protocol response.
+#[derive(Clone, Debug, PartialEq)]
+pub enum MemcachedValue {
```

Comment:
This should be renamed to something else so as not to create confusion with `AsMemcachedValue`.  Perhaps `MemcachedResponse` or something similar.